Initial draft for mTLS encryption annoucement and deep dive architecture (#939)
cilium/cilium.io: src/posts/2026-02-24-launch-of-mTLSencryption/index.md

---
path: /blog/2026/03/23/native-mtls-cilium
date: 2026-03-23T09:00:00.000Z
title: "Native mTLS for Cilium: Transparent Encryption Meets Cloud Native Identity"

> isFeatured: false

Reviewer: ```suggestion
isFeatured: true
```

categories:
  - Technology
tags:
  - Cilium
  - mTLS
  - Security
  - Encryption
  - Service Mesh
  - eBPF
  - SPIRE
  - SPIFFE
  - KubeCon
  - CNCF
---

# Native mTLS for Cilium: Transparent Encryption Meets Cloud Native Identity

*Announcing native mutual TLS in the Cilium datapath — zero-drop, inline encryption with Kubernetes-native identity, no sidecars required.*

![Native mTLS for Cilium](native-mtls-cover.png)

Today at [KubeCon + CloudNativeCon Europe 2026](https://events.linuxfoundation.org/kubecon-cloudnativecon-europe/) in Amsterdam, we are excited to announce **native mTLS support in Cilium** — a major step forward for transparent, zero-trust encryption in Kubernetes. This capability brings inline mutual TLS directly into the Cilium datapath, unifying authentication and encryption into a single operation. No sidecars, no dropped first packets, and no application changes required.

Mutual TLS has long been the gold standard for workload-to-workload security in cloud native environments, but achieving it has historically meant accepting significant operational complexity. With this release, Cilium eliminates that tradeoff.

> 

Reviewer: ```suggestion
# How Open Source Drives Innovation: Lessons from the Evolution of the Service Mesh

In the Cilium project, we’ve always believed that open source isn’t just a way to share code, it’s also a way to drive innovation. By putting ideas into the public domain, engineers learn from them, challenge them, and build on them to drive progress forward. The evolution of the service mesh towards ztunnel is a perfect case study of this process in action. Projects like Cilium, Istio, and Linkerd have each contributed unique approaches, learning from one another to enhance performance, scalability, and user experience.

# Different Paths, Same Destination

When the service mesh pattern first emerged, sidecars were the dominant model. Projects like Istio and Linkerd standardized on injecting a proxy alongside each application instance, offering consistent L7 observability, security, and traffic management. But as these meshes grew in adoption, so did the operational complexity and performance overhead of sidecar-heavy architectures.

Interestingly, Linkerd actually started with a per-node proxy model in its earliest versions. The idea was to avoid the complexity of managing sidecars altogether. However, they encountered real limitations which led them to pivot toward a sidecar-based approach. It was a pragmatic move: sidecars gave them per-pod context and control, which was hard to achieve with the technologies available at the time.

# Cilium: Rethinking Service Mesh with eBPF

Fast forward a few years, and the landscape had changed. Cilium introduced a sidecarless service mesh built on eBPF, which enabled per-pod visibility, identity, and policy enforcement, all without the need for injected sidecar proxies. With eBPF running directly in the Linux kernel, Cilium was able to revisit the original promise of node-local simplicity while delivering per-pod context and control.

This was a major inflection point. By removing the operational burden of sidecars while maintaining strong security and observability guarantees, Cilium opened the door to a new generation of service mesh architectures that removed the need for a sidecar injected into each and every pod.

# Enter ztunnel

Cilium’s introduction of the sidecarless approach led to months of debate and discussion about the pros and cons. In our view, healthy discussion and probing different technical positions can lead to real progress. The following year, Istio Ambient was announced, proposing a new data plane model for Istio that eliminates sidecars in favor of a per-node ztunnel component for Layer 4 security, and an optional Layer 7 proxy layer for application traffic control.

[Performance results](https://arxiv.org/pdf/2411.02267) for Ambient are promising—lower CPU usage, faster pod starts, and improved network efficiency compared to Istio’s traditional sidecar deployments. But what really piqued our interest is that for encrypted TCP connections, the ztunnel approach even out-performs Cilium. 

One of the main reasons for ztunnel’s better performance is that by encrypting a TCP payload stream, it can avoid the per-packet encryption overhead inherent in Cilium’s use of WireGuard/IPsec.

But this performance boost does come with a few trade-offs.

One area to consider is upgrades. Cilium’s eBPF programs can be left in the kernel during a control plane upgrade, allowing data to continue flowing while user space components are replaced or restarted. With ztunnel, any TCP connections will need to be torn down during upgrade.

Another factor, which may be important to some users, is that WireGuard and IPsec can be used to encrypt any IP traffic, whereas ztunnel is limited to TCP traffic.

# Bringing ztunnel to Cilium

When we started looking at [bringing ztunnel support into Cilium](https://github.com/cilium/cilium/issues/38548), it seemed like we could combine the best of both worlds. The top-tier performance of ztunnel, combined with Cilium’s ease of policy control, and the ability to support all types of IP traffic.

Bringing ztunnel support into Cilium brings native mTLS to the data path and enables enhanced performance for ztunnel encrypted connections.
```

> ## The Challenge: Authentication Without Encryption

Reviewer: I don't know of anyone doing Authentication Without Encryption. The more common complaint is that it wasn't session based authentication and for instance you couldn't do pod to pod encryption on the same node (not that that really helps security wise)

    |                 |   lookup -> MISS    |                       |
    |                 |                     |                       |
    |   X DROPPED     |-- 4. Signal ----->>|                       |
    |                 |                     |-- 5. Get SPIFFE cert  |
    |                 |                     |   from SPIRE          |
    |                 |                     |== 6. TLS 1.3 ======>> |
    |                 |                     |   (out-of-band)       |
    |                 |                     |   7. Validate cert  <<|
    |                 |<<-- 8. Write -------|                       |
    |                 |   auth_map entry    |                       |

                     SUBSEQUENT PACKETS (ALLOWED, BUT PLAINTEXT)

  Pod A          BPF Datapath (A)                      BPF Datapath (B)    Pod B
    |                 |                                       |               |
    |-- 9. Packet --> |                                       |               |
    |                 |-- auth_map lookup -> HIT              |               |
    |                 |========= PLAINTEXT ================>> |-- forward --> |
```

> While this approach is elegant for pure identity verification, it leaves a critical gap in environments that require confidentiality guarantees — which is, increasingly, every production Kubernetes cluster.

Reviewer: ```suggestion
```

## Introducing Native mTLS: Ztunnel Integration

> Cilium's native mTLS is powered by a [forked version of ztunnel](https://github.com/cilium/ztunnel) — a lightweight, per-node Rust proxy originally from Istio's ambient mesh — reengineered to integrate directly with the Cilium control plane and **SPIRE** as the certificate authority. We introduced native SPIRE support into ztunnel ([cilium/ztunnel#4](https://github.com/cilium/ztunnel/pull/4)), replacing the upstream CA dependency with SPIRE's Delegated Identity API. SPIRE is the **only** CA mode supported in Cilium's mTLS solution, providing a production-grade, [CNCF-graduated](https://www.cncf.io/projects/spiffe/) identity foundation.

Reviewer: ```suggestion
Cilium's native mTLS is powered by a [forked version of ztunnel](https://github.com/cilium/ztunnel) — a lightweight, per-node Rust proxy originally from Istio's ambient mesh — reengineered to integrate directly with the Cilium control plane and **SPIRE** as the certificate authority. We introduced native SPIRE support into ztunnel ([cilium/ztunnel#4](https://github.com/cilium/ztunnel/pull/4)), replacing the upstream CA dependency with SPIRE's Delegated Identity API. SPIRE is the **only** CA mode supported in Cilium's mTLS solution, providing a production-grade, [CNCF-graduated](https://www.cncf.io/projects/spiffe/) identity foundation.

While we are currently maintaining a fork to accelerate these capabilities, our goal is to align with and eventually upstream these improvements to the broader ecosystem.
```

```

When a namespace is enrolled, the operator queries all service accounts in that namespace and creates corresponding SPIRE registration entries with SPIFFE IDs in the format `spiffe://<trust-domain>/ns/<namespace>/sa/<service-account>`, parented under `/ztunnel`. When a namespace is un-enrolled, those entries are cleaned up.

## Enrollment Model: Gradual Rollout by Namespace

A key design decision is that mTLS enrollment is **per-namespace**, controlled by a single label:

```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: production
  labels:
    io.cilium/mtls-enabled: "true"
```

This gives platform teams a natural rollout mechanism: enable mTLS for individual namespaces as they are ready, without affecting the rest of the cluster. Mixed traffic scenarios work predictably:

| Source | Destination | Result |
|--------|-------------|--------|
| Enrolled | Enrolled | **Encrypted** (mTLS over HBONE) |
| Enrolled | Non-enrolled | Plaintext passthrough |
| Non-enrolled | Enrolled | Plaintext (captured by ztunnel, but not encrypted) |
| Non-enrolled | Non-enrolled | Normal Cilium datapath (no ztunnel involvement) |

> Encryption only occurs when **both** pods are enrolled. Mixed traffic degrades gracefully to plaintext — no breakage, no hard failures.

Reviewer: ```suggestion
Encryption only occurs when **both** pods are enrolled. Mixed traffic degrades gracefully to plaintext without breakage or hard failures.
```

## Why This Matters

> ### 1. Encryption by Default, Without Operational Overhead

Reviewer: ```suggestion
### 1. mTLS by Default, Without Operational Overhead
```

### 2. No Dropped Packets

Unlike the existing mutual auth approach, ztunnel holds connections inline while the mTLS tunnel is established. There is no "first packet drop" penalty — connections succeed on the first attempt.

### 3. Kubernetes-Native Identity

Workload identity is expressed as `spiffe://<trust-domain>/ns/<namespace>/sa/<service-account>`, aligning directly with Kubernetes RBAC primitives. This is more natural and auditable than Cilium's numeric identity model used by legacy mutual auth.

### 4. Production-Grade Identity with SPIRE

Workload certificates are managed by [SPIRE](https://spiffe.io/), the CNCF-graduated reference implementation of the SPIFFE standard. SPIRE provides automatic certificate rotation, workload attestation via Kubernetes node and pod selectors, and a trust chain rooted in a dedicated SPIRE server. The Cilium operator automates SPIRE registration entry management — when a namespace is enrolled, all service account identities are automatically registered with SPIRE.

> ### 5. Ecosystem Convergence

Reviewer: ```suggestion
### 5. Best of Both Worlds: Performance & Flexibility

By encrypting a TCP payload stream, ztunnel can avoid the per-packet encryption overhead inherent in IPsec. This provides a significant performance boost for TCP-heavy workloads. Meanwhile, users requiring encryption for non-TCP traffic or hitless upgrades can still opt for Cilium’s WireGuard or IPsec modes.

### 6. Ecosystem Convergence
```

| **Identity**         | `spiffe://.../identity/<num-id>`  | `spiffe://.../ns/<ns>/sa/<sa>`   |
| **Enrollment**       | Per CiliumNetworkPolicy rule      | Per namespace label              |
| **Cert source**      | SPIRE only                        | SPIRE (automated registration)   |
| **Granularity**      | Per identity pair (BPF cache)     | Per workload pair (H2 mux)       |
| **App changes**      | None                              | None                             |

## Getting Started

Enabling native mTLS requires three things:

1. **Enable the ztunnel feature** in your Cilium Helm values:

```yaml
ztunnel:
  enabled: true
```

2. **Deploy ztunnel** as a DaemonSet (managed via Helm).

3. **Enroll namespaces** by adding the label:

```bash
kubectl label namespace <name> io.cilium/mtls-enabled=true
```

That's it. All pods in enrolled namespaces will automatically have their traffic encrypted with mTLS. You can verify the status with:

```bash
cilium status
```

The ztunnel component will appear in the output, showing enrollment state and certificate health.

## A Joint Effort: Microsoft and Isovalent

> Native mTLS in Cilium is the product of a deep, joint engineering collaboration between **Microsoft** and **Isovalent** (the creators of Cilium, now part of Cisco). Microsoft's team has been given **maintainership of the ztunnel feature** in the upstream Cilium codebase, a reflection of the shared commitment to building this capability in the open, for the entire community.

Reviewer: ```suggestion
Native mTLS in Cilium is the product of a deep, joint engineering collaboration between **Microsoft** and **Isovalent** (the creators of Cilium, now part of Cisco). Microsoft's team has been **the driving force of the ztunnel feature** in the upstream Cilium codebase, a reflection of the shared commitment to building this capability in the open, for the entire community.
```

The work spans multiple areas:

**Core control plane** — merged upstream into Cilium:
- Remote workload awareness ([#41945](https://github.com/cilium/cilium/pull/41945))
- ZDS server for pod enrollment ([#42364](https://github.com/cilium/cilium/pull/42364))
- Namespace-based mTLS enrollment ([#41944](https://github.com/cilium/cilium/pull/41944))
- XDS namespace filtering ([#43128](https://github.com/cilium/cilium/pull/43128))
- `cilium status` integration ([#43227](https://github.com/cilium/cilium/pull/43227))

**Datapath and deployment** — enabling production readiness:
- Idempotent in-pod iptables rules ([#42122](https://github.com/cilium/cilium/pull/42122))
- go-iptables library migration ([#42160](https://github.com/cilium/cilium/pull/42160))
- Helm-managed ztunnel DaemonSet ([#43763](https://github.com/cilium/cilium/pull/43763))
- Configurable SPIRE client ([#44136](https://github.com/cilium/cilium/pull/44136))
- Namespace enrollment reconciler for SPIRE ([#44275](https://github.com/cilium/cilium/pull/44275))

**Ztunnel fork** ([cilium/ztunnel](https://github.com/cilium/ztunnel)) — native SPIRE support:

> A key piece of this collaboration was introducing native SPIRE support directly into ztunnel. Upstream ztunnel was designed exclusively for Istio's control plane and its built-in CA (istiod) — it had no concept of SPIRE as a certificate authority. We introduced SPIRE's Delegated Identity API into ztunnel ([#4](https://github.com/cilium/ztunnel/pull/4)), enabling ztunnel to obtain X.509-SVIDs directly from the local SPIRE agent via workload attestation. This includes PID-based attestation where each workload is individually verified by its container process ID, providing stronger security guarantees than metadata-only approaches. SPIRE is the **only** CA mode supported in Cilium's mTLS solution — there is no fallback to Istio's CA.

Reviewer: ```suggestion
A key piece of this collaboration was introducing native SPIRE support directly into ztunnel. Upstream ztunnel was designed exclusively for Istio's control plane and its built-in CA (istiod) — it had no concept of SPIRE as a certificate authority. We introduced SPIRE's Delegated Identity API into ztunnel ([#4](https://github.com/cilium/ztunnel/pull/4)), enabling ztunnel to obtain X.509-SVIDs directly from the local SPIRE agent via workload attestation. This includes PID-based attestation where each workload is individually verified by its container process ID, providing stronger security guarantees than metadata-only approaches. SPIRE is the **only** CA mode supported in Cilium's mTLS solution — there is no fallback to Istio's CA. We look forward to contributing these changes back to the upstream ztunnel project to ensure long-term ecosystem alignment.
```

- Unix socket support for XDS ([#1](https://github.com/cilium/ztunnel/pull/1))
- Native SPIRE workload attestation via Delegated Identity API ([#4](https://github.com/cilium/ztunnel/pull/4))
- CI: container image build pipeline ([#7](https://github.com/cilium/ztunnel/pull/7))

**Testing** — comprehensive validation:
- Encryption connectivity tests ([#43229](https://github.com/cilium/cilium/pull/43229))
- Feature checks and e2e scenarios ([#43255](https://github.com/cilium/cilium/pull/43255), [#43803](https://github.com/cilium/cilium/pull/43803))

> 

Reviewer: ```suggestion

# Conclusion: Open Source Drives Us All Forward

The journey to native mTLS in Cilium is a perfect case study in how open source drives innovation. By putting ideas into the public domain, engineers learn from, challenge, and build upon them. The evolution of the service mesh has seen many paths, from the early sidecar models of Istio and Linkerd to Cilium’s eBPF-powered sidecarless approach. Each iteration has been a response to the community's need for better performance and lower operational complexity.

When Istio Ambient introduced ztunnel, it offered a new data plane model that offered new efficiency gains in encrypting TCP payload streams. By integrating ztunnel support into Cilium, we are embracing the "best of both worlds." We combine the top-tier performance of ztunnel for TCP and native mTLS with Cilium’s robust policy control and the flexibility of WireGuard/IPsec for non-TCP traffic and hitless upgrades.

This integration is the result of a healthy feedback loop where great ideas are shared, refined, and reimagined. At Isovalent and Microsoft, we are proud to be part of a community where learning goes both ways. By converging the innovations of Cilium, Istio, and SPIRE, we ensure that the ecosystem remains modular, resilient, and focused on the user.

Because when open source wins, _users win_.
```

Interestingly, Linkerd actually started with a per-node proxy model in its earliest versions. The idea was to avoid the complexity of managing sidecars altogether. However, they encountered real limitations which led them to pivot toward a sidecar-based approach. It was a pragmatic move: sidecars gave them per-pod context and control, which was hard to achieve with the technologies available at the time.

# Cilium: Rethinking Service Mesh with eBPF

Fast forward a few years, and the landscape had changed. Cilium introduced a sidecarless service mesh built on eBPF, which enabled per-pod visibility, identity, and policy enforcement, all without the need for injected sidecar proxies. With eBPF running directly in the Linux kernel, Cilium was able to revisit the original promise of node-local simplicity while delivering per-pod context and control.

This was a major inflection point. By removing the operational burden of sidecars while maintaining strong security and observability guarantees, Cilium opened the door to a new generation of service mesh architectures that removed the need for a sidecar injected into each and every pod.

# Enter ztunnel

> Cilium’s introduction of the sidecarless approach led to months of debate and discussion about the pros and cons. In our view, healthy discussion and probing different technical positions can lead to real progress. The following year, Istio Ambient was announced, proposing a new data plane model for Istio that eliminates sidecars in favor of a per-node ztunnel component for Layer 4 security, and an optional Layer 7 proxy layer for application traffic control.

Reviewer: ```suggestion
Cilium’s introduction of the sidecarless approach led to months of debate and discussion about the pros and cons. In our view, healthy discussion and probing different technical positions can lead to real progress. The following year, Istio Ambient was announced (listen to an interview with the authors on why they created it [here](https://kubernetespodcast.com/episode/189-ambient-mesh/)), proposing a new data plane model for Istio that eliminates sidecars in favor of a per-node ztunnel component for Layer 4 security, and an optional Layer 7 proxy layer for application traffic control. 
```

---
path: /blog/2026/03/23/native-mtls-cilium
date: 2026-03-23T09:00:00.000Z

> title: "Native mTLS for Cilium: Transparent Encryption Meets Cloud Native Identity"

Reviewer: ```suggestion
title: "Native mTLS for Cilium: Transparent Encryption Meets Cloud Native Identity with ztunnel"
```

isFeatured: false
ogImage: native-mtls-cover.png

> ogSummary: "Cilium now brings native, inline mTLS to the Kubernetes datapath — unifying authentication and encryption in a single step, with zero application changes."

Reviewer: ```suggestion
ogSummary: "Cilium now brings native, inline mTLS to the Kubernetes datapath with ztunnel, unifying authentication and encryption in a single step, with zero application changes."
```

# How Open Source Drives Innovation: Lessons from the Evolution of the Service Mesh

In the Cilium project, we’ve always believed that open source isn’t just a way to share code, it’s also a way to drive innovation. By putting ideas into the public domain, engineers learn from them, challenge them, and build on them to drive progress forward. The evolution of the service mesh towards ztunnel is a perfect case study of this process in action. Projects like Cilium, Istio, and Linkerd have each contributed unique approaches, learning from one another to enhance performance, scalability, and user experience.

# Different Paths, Same Destination

When the service mesh pattern first emerged, sidecars were the dominant model. Projects like Istio and Linkerd standardized on injecting a proxy alongside each application instance, offering consistent L7 observability, security, and traffic management. But as these meshes grew in adoption, so did the operational complexity and performance overhead of sidecar-heavy architectures.

> Interestingly, Linkerd actually started with a per-node proxy model in its earliest versions. The idea was to avoid the complexity of managing sidecars altogether. However, they encountered real limitations which led them to pivot toward a sidecar-based approach. It was a pragmatic move: sidecars gave them per-pod context and control, which was hard to achieve with the technologies available at the time.

Reviewer: ```suggestion
Interestingly, Linkerd actually started with a per-node proxy model in its earliest versions. However, they encountered real limitations which led them to pivot toward a sidecar-based approach. It was a pragmatic move: sidecars gave them per-pod context and control, which was hard to achieve with the technologies available at the time.
```

[Performance results](https://arxiv.org/pdf/2411.02267) for Ambient are promising—lower CPU usage, faster pod starts, and improved network efficiency compared to Istio’s traditional sidecar deployments. But what really piqued our interest is that for encrypted TCP connections, the ztunnel approach even out-performs Cilium. 

One of the main reasons for ztunnel’s better performance is that by encrypting a TCP payload stream, it can avoid the per-packet encryption overhead inherent in Cilium’s use of WireGuard/IPsec.

But this performance boost does come with a few trade-offs.

One area to consider is upgrades. Cilium’s eBPF programs can be left in the kernel during a control plane upgrade, allowing data to continue flowing while user space components are replaced or restarted. With ztunnel, any TCP connections will need to be torn down during upgrade.

> Another factor, which may be important to some users, is that WireGuard and IPsec can be used to encrypt any IP traffic, whereas ztunnel is limited to TCP traffic.

Reviewer: ```suggestion
Another factor, which may be important to some users, is that WireGuard and IPsec can be used to encrypt any IP traffic, whereas ztunnel is limited to TCP traffic for now.
```

# Bringing ztunnel to Cilium

When we started looking at [bringing ztunnel support into Cilium](https://github.com/cilium/cilium/issues/38548), it seemed like we could combine the best of both worlds. The top-tier performance of ztunnel, combined with Cilium’s ease of policy control, and the ability to support all types of IP traffic.

Bringing ztunnel support into Cilium brings native mTLS to the data path and enables enhanced performance for ztunnel encrypted connections.

## The Challenge: Network-Level Authentication, Not Session-Based

Cilium's [existing mutual authentication](https://isovalent.com/blog/post/2022-05-03-servicemesh-security/) capability (introduced as beta in Cilium 1.14) was a significant innovation, using eBPF to verify workload identity at the datapath level. However, it had some important limitations:

> - **Network-level, not session-based authentication.** The mutual authentication handshake had nodes authenticate each other and then encapsulate all network traffic to build an authenticated and encrypted virtual network. This meant authentication happened at the node level rather than on a per-session basis — so, for example, pod-to-pod encryption on the same node was not possible (though arguably this has limited security benefit in practice).

Reviewer: ```suggestion
- **Network-level, not session-based authentication.** The mutual authentication handshake had nodes authenticate each other and then encapsulate all network traffic to build an authenticated and encrypted virtual network. This meant authentication happened at the node level rather than on a per-session basis — so, for example, pod-to-pod encryption on the same node was not possible.
```

    |                 |                     |   from SPIRE          |
    |                 |                     |== 6. TLS 1.3 ======>> |
    |                 |                     |   (out-of-band)       |
    |                 |                     |   7. Validate cert  <<|
    |                 |<<-- 8. Write -------|                       |
    |                 |   auth_map entry    |                       |

                     SUBSEQUENT PACKETS (ALLOWED, BUT PLAINTEXT)

  Pod A          BPF Datapath (A)                      BPF Datapath (B)    Pod B
    |                 |                                       |               |
    |-- 9. Packet --> |                                       |               |
    |                 |-- auth_map lookup -> HIT              |               |
    |                 |========= PLAINTEXT ================>> |-- forward --> |
```

While this approach is elegant for pure identity verification, it leaves a critical gap in environments that require confidentiality guarantees — which is, increasingly, every production Kubernetes cluster.

## Introducing Native mTLS: Ztunnel Integration

> Cilium's native mTLS is powered by a [forked version of ztunnel](https://github.com/cilium/ztunnel) — a lightweight, per-node Rust proxy originally from Istio's ambient mesh — reengineered to integrate directly with the Cilium control plane and **SPIRE** as the certificate authority. We introduced native SPIRE support into ztunnel ([cilium/ztunnel#4](https://github.com/cilium/ztunnel/pull/4)), replacing the upstream CA dependency with SPIRE's Delegated Identity API. SPIRE is the **only** CA mode supported in Cilium's mTLS solution, providing a production-grade, [CNCF-graduated](https://www.cncf.io/projects/spiffe/) identity foundation.

Reviewer: ```suggestion
Cilium's native mTLS is powered by [ztunnel](https://github.com/cilium/ztunnel) — a lightweight, per-node Rust proxy originally from Istio's ambient mesh — reengineered to integrate directly with the Cilium control plane and **SPIRE** as the certificate authority. We introduced native SPIRE support into ztunnel ([cilium/ztunnel#4](https://github.com/cilium/ztunnel/pull/4)), replacing the upstream CA dependency with SPIRE's Delegated Identity API. SPIRE is the **only** CA mode supported in Cilium's mTLS solution, providing a production-grade, [CNCF-graduated](https://www.cncf.io/projects/spiffe/) identity foundation.
```
